core/primitives-core/src/serialize.rs

```diff
@@ -212,3 +216,27 @@ pub mod option_u128_dec_format {
         }
     }
 }
+
+#[derive(Debug, Deserialize)]
+pub struct DeserializedWithDebug<T>(pub T);
+
+impl<T> DeserializedWithDebug<T> {
+    fn _deserialize<'de, D>(deserializer: D) -> Result<T, D::Error>
+    where
+        T: Deserialize<'de>,
+        D: Deserializer<'de>,
+    {
+        if cfg!(debug_assertions) {
```

Comment:
`debug_assertions` will largely be disabled 99/100 times, as running a non-optimized near code is not all that feasible, and rarely anybody will go out of their way to set `-Cdebug-assertions=yes`.